src/VImageCleaning.cpp

```diff
@@ -424,6 +424,98 @@ bool VImageCleaning::InitNNImageCleaning()
 	return true;
 }
 
+/*
+ * Calculate averaged IPR graph over all telescopes when there is not enough statistics
+ * Call it if flag fIPRAverageTel > 0.
+ */
+TGraphErrors* VImageCleaning::calculateIPRGraphsAveragedAllTelescopes( unsigned int iSummationWindow )
+{
+
+	cout << "VImageCleaning::calculateIPRGraphsAveragedAllTelescopes: doing average of IPR graphs." << endl;
+	TGraphErrors* IPRGraphAverageTel = new TGraphErrors();
+	
+	char InitTelID = fData->getTelID();
+	
+	for( unsigned int fTelescope = 0; fTelescope < fData->getRunParameter()->fNTelescopes; fTelescope++ )
+	{
+		//change it back to original tel ID
+		fData->setTelID( fTelescope );
+		cout << "VImageCleaning::calculateIPRGraphsAveragedAllTelescopes: getting IPR for telescope: ";
+        cout << fTelescope << " " << fData->getDetectorGeometry()->getTelID() << endl;
+		TGraphErrors* g = fData->getIPRGraph( iSummationWindow, false );
+		if( !g )
+		{
+			continue;
+		}
+        cout << "IPR graph: " << g->GetName() << endl;
+		g->Sort();
+		
+		int NPointsAverageTel = IPRGraphAverageTel->GetN();
+		float eps = 1e-3;
+		for( int k = 0; k < g->GetN(); k++ )
+		{
+			double XIndivTel =  g->GetPointX( k );
+			double YIndivTel =  g->GetPointY( k );
+			
+			double XEIndivTel = g->GetErrorX( k );
+			double YEIndivTel = g->GetErrorY( k );
+			int i_ocurrence = 0;
+			if( fTelescope == 0 )
```

Comment:
I've changed the logic to start the if..else with the first case (fTelescope == 0 ). But note that this is not working for 3-telescope runs, when no T1 exists (T234 runs). Maybe you can find a better solution to have it more flexible.

Reply:
Fixed by replacing 4 by getNTel() in all occurrences.

Reply:
As above, use `fData->getTeltoAna().size()`.